spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/aggregation/GeoNearOperation.java

```diff
@@ -28,11 +28,15 @@
 public class GeoNearOperation implements AggregationOperation {
 
 	private final NearQuery nearQuery;
+	private final String distanceField;
 
-	public GeoNearOperation(NearQuery nearQuery) {
+	public GeoNearOperation(NearQuery nearQuery, String distanceField) {
```

Comment:
I think we have to overload the constructor to not break existing clients using the old constructor. I suggest to add a hint to favor `Aggregation.geoNear(…)` over using the constructors directly. Should we actually do that for all of the operation constructors that actually have a static factory method equivalent in `Aggregation`?